### DISCUSSION TITLE
The error occurs when I only have 2 consumers but 3 messages arrive at the same time

### DISCUSSION THREAD

**Discussion by MinhSuIt** [rabbitmq/rabbitmq-server#11275] [Questions] [answered]
How to make the 3rd message wait until the previous 2 messages are processed?
God bless :)

**Comment by michaelklishin** [comment, marked as answer]
No one can provide an informed answer since you haven't posted any code that can be used to reproduce.

Most likely [tutorial 2](https://www.rabbitmq.com/tutorials) and [channel prefetch](https://www.rabbitmq.com/docs/confirms#channel-qos-prefetch) are highly relevant to what you are trying to do.
